### ISSUE TITLE
What happened to this core? (regression)

### ISSUE THREAD

**Issue by veganvelociraptor** [libretro/virtualjaguar-libretro#72]
I remember this core was perfectly usable, and almost all games were fully playable.

Something must've happened a while back, I thought for a long time it was something wrong in the core (Linux, x64) I compiled myself. But now I've tried out the core on the latest Lakka on Rpi 3 and 4, and also Retroarch on an Nvidia Shield with pre-compiled cores and they all suffer the same slowdowns.

No games I've tried are playable. Not even the BIOS animation is smooth, regardless if the fast blitter hack is switched on or off.

Since this core seems more or less abandoned, does anyone keep track of the last working version and knows if it still works with an up-to-date Retroarch version?

**Comment by hizzlekizzle** [comment]
Since you can compile, would you mind bisecting to find out when it changed?

**Comment by veganvelociraptor** [comment]
Hence my question, I was asking so I wouldn't have to as it takes time.

**Comment by hizzlekizzle** [comment]
I just compiled a bunch of commits stretching back to 2015 and the speed seems to fluctuate slightly, but nothing major here. I got ~115 fps in 2015, up to ~140 fps in 2017 and back down to ~130 now.

**Comment by veganvelociraptor** [comment]
I just tried out Cannon Fodder and RetroArch reports a consistent 30-something fps, but the game stutters so badly it looks more like a slideshow of maybe 1-3 fps.

This is when tested on my main rig with a Ryzen 2600x CPU, AMD 5700 gfxcard and 32 gb of ram. But the virtual jaguar core has the exact issues on my Nvidia shield. And with all games tested, not only Cannon Fodder.

**Comment by hizzlekizzle** [comment]
Did you identify a time/commit when that performance was different/better? Have you tried it with default settings to ensure nothing is conflicting?

**Comment by veganvelociraptor** [comment]
I haven't had time to do retro gaming for a while now, maybe 2 years. Last I remember, I was using Lakka v2.3.x where it worked fine. I didn't notice this problem until upgrading to Lakka v3.x years later and installing Retroarch on my main Linux rig where this issue also occurs.

Gonna try out a clean install on a separate computer, as @hizzlekizzle suggested, with default settings.

**Comment by JoeMatt** [comment]
Different ROMs show different performance with the different blitter and BIOS options for one.
Second, if using a BIOS, check the version used. There are some glitchy BIOSs out there too or ones meant for debugging.

I only ever test on Intel and Apple Silicon/arm. I've used VJ on RPI a few times but not enough to remember what performance was like.

**Comment by JoeMatt** [comment]
I tagged this waiting for test by @veganvelociraptor until further notice.

**Comment by JoeMatt** [comment]
Tested myself same game on retroarch-metal for M1 macOS. Ran smooth here.

**Comment by veganvelociraptor** [comment]
Thanks guys for being patient with me. I've been "busy" since my Steam Deck arrived.

However, it did present itself with an opportunity to test this core out on a new device. I installed the Steam version of Retroarch and then proceeded to manually install the Virtual Jaguar core since its not included as a DLC.

Only changes I made was changing the display driver to Vulcan, and everything runs smoothly on the Steam Deck. Including the BIOS animation which is little more than a slideshow with stuttering audio on my other devices.

Next thing I'll try out is see if various changes in settings will make it run as slow as on my other devices (shaders etc), and then try out the nightly release of Retroarch incl cores.
